src/EditorFeatures/Core/SymbolSearch/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -110,19 +104,19 @@ public Updater(SymbolSearchUpdateEngine service, string source, string localSett
                 }
 
                 // Keep on looping until we're told to shut down.
-                while (!_service._updateCancellationToken.IsCancellationRequested)
+                while (!cancellationToken.IsCancellationRequested)
                 {
-                    await _service.LogInfoAsync("Starting update").ConfigureAwait(false);
+                    await _service.LogInfoAsync("Starting update", cancellationToken).ConfigureAwait(false);
                     try
                     {
-                        var delayUntilNextUpdate = await UpdateDatabaseInBackgroundWorkerAsync().ConfigureAwait(false);
+                        var delayUntilNextUpdate = await UpdateDatabaseInBackgroundWorkerAsync(cancellationToken).ConfigureAwait(false);
 
-                        await _service.LogInfoAsync($"Waiting {delayUntilNextUpdate} until next update").ConfigureAwait(false);
-                        await Task.Delay(delayUntilNextUpdate, _service._updateCancellationToken).ConfigureAwait(false);
+                        await _service.LogInfoAsync($"Waiting {delayUntilNextUpdate} until next update", cancellationToken).ConfigureAwait(false);
+                        await Task.Delay(delayUntilNextUpdate, cancellationToken).ConfigureAwait(false);
                     }
                     catch (OperationCanceledException)
                     {
-                        await _service.LogInfoAsync("Update canceled. Ending update loop").ConfigureAwait(false);
+                        await _service.LogInfoAsync("Update canceled. Ending update loop", cancellationToken).ConfigureAwait(false);
```

Comment:
this is the part that is now broken.  Cancellation tells us to shutdown and to stop updating this specific index.